trace-obfuscation/src/redis_tokenizer.rs

```diff
@@ -0,0 +1,443 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2023-Present Datadog, Inc.
+
+#[derive(Debug)]
+pub enum RedisTokenType {
+    RedisTokenCommand,
+    RedisTokenArgument,
+}
+
+pub struct RedisTokenizer {
+    data: Vec<char>,
+    cur_char: char,
+    offset: Option<usize>,
```

Comment:
What do you think about removing this option, setting 0 instead of None. I feel that the code would be simpler but I may have missed something?

Reply:
The main purpose was so that `next()` knows whether or not it's being called for the very first time. Alternatively, I could add a `has_started` field, and remove the option?